src/root/components/OperationalUpdatePdfDocument/index.tsx

```diff
@@ -0,0 +1,874 @@
+  const defaultValue = '-';
+
+  if (isValidNumber(value)) {
+    const formattedNumber = addSeparator(value) ?? defaultValue;
+
+    if (prefix) {
+      return `${prefix}${formattedNumber}`;
+    }
+
+    return formattedNumber;
+  }
+
+  return defaultValue;
+}
+
+function TextOutput(props: {
+  label: string;
+  value?: string;
+  columns?: '1/3' | '2/3' | '3/3' | '1/2';
+}) {
+  const {
+    label,
+    value,
+    columns = '1/3',
+  } = props;
+
+  const styleMap = {
+    '1/3': pdfStyles.oneByThree,
+    '2/3': pdfStyles.twoByThree,
+    '3/3': pdfStyles.threeByThree,
+    '1/2': pdfStyles.oneByTwo,
+  };
+
+  return (
+    <View
+      style={[
+        pdfStyles.textOutput,
+        styleMap[columns] ?? pdfStyles.oneByThree,
+      ]}
+    >
+      <Text style={pdfStyles.textOutputLabel}>
+        {label}
+      </Text>
+      <Text style={pdfStyles.textOutputValue}>
+        {value}
+      </Text>
+    </View>
+  );
+}
+
+interface DrefOptions {
+  disaster_category: NumericKeyValuePair[];
+  national_society_actions: StringKeyValuePair[];
+  needs_identified: StringKeyValuePair[];
+  planned_interventions: StringKeyValuePair[];
+  status: NumericKeyValuePair[];
+  type_of_onset: NumericKeyValuePair[];
+  users: {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+    username: string;
+  }[];
+}
+
+interface Props {
+  operationalUpdateResponse: DrefOperationalUpdateApiFields;
+  drefOptions: DrefOptions;
+  // onLoad?: (params: { blob?: Blob }) => void;
+  strings: Strings;
+}
+
+function OperationalUpdatePdfDocument(props: Props) {
+  const {
+    operationalUpdateResponse,
+    drefOptions,
+    strings,
+  } = props;
+
+  const getMaps = () => {
+    if (!drefOptions) {
+      return [
+        {},
+        {},
+        {},
+      ];
+    }
+
+    return [
+      listToMap(drefOptions.planned_interventions, d => d.key, d => d.value),
+      listToMap(drefOptions.needs_identified, d => d.key, d => d.value),
+      listToMap(drefOptions.national_society_actions, d => d.key, d => d.value),
+    ];
+  };
+  const [
+    piMap,
+    niMap,
+    nsaMap,
+  ] = getMaps();
+
+  const getAffectedAreas = () => {
+    if (operationalUpdateResponse?.country_district) {
+      let areas = '';
+      const districts = operationalUpdateResponse.country_district.map(d => d.district_details);
+
+      districts.forEach(d => {
+        const names = d.map(dd => dd.name);
+        areas += names.join(', ');
+      });
+
+      return areas;
+    }
+
+    return undefined;
+  };
+  const affectedAreas = getAffectedAreas();
+
+  const logoUrl = '/assets/graphics/layout/go-logo-2020.png';
+
+  const isImminentOnset = operationalUpdateResponse?.disaster_type === ONSET_IMMINENT;
+  const documentTitle = [
+    operationalUpdateResponse?.country_district.map(d => d.country_details?.name).join(', '),
+    operationalUpdateResponse?.title
+  ].join(' | ');
```

Comment:
use useMemo

Reply:
This would be resolved to string, so `useMemo` is not actually necessary here